src/plugins/template/src/template_compiled_model.cpp

```diff
@@ -0,0 +1,247 @@
+        input_info->setPrecision(td.getPrecision());
+    }
+}
+void fill_output_info(const ov::Output<ov::Node>& output, InferenceEngine::DataPtr& output_info) {
+    if (!output_info) {
+        // Create input info
+        const auto& res_name = ov::op::util::create_ie_output_name(output);
+        auto dims = get_dims(output);
+        InferenceEngine::TensorDesc desc(InferenceEngine::details::convertPrecision(output.get_element_type()),
+                                         dims,
+                                         InferenceEngine::TensorDesc::getLayoutByDims(dims));
+        output_info = std::make_shared<InferenceEngine::Data>(res_name, desc);
+    }
+    auto& rt_info = output.get_rt_info();
+    auto it = rt_info.find("ie_legacy_td");
+    if (it != rt_info.end()) {
+        auto td = it->second.as<InferenceEngine::TensorDesc>();
+        output_info->reshape(td.getDims(), td.getLayout());
+        output_info->setPrecision(td.getPrecision());
+    }
+}
+}  // namespace legacy_convert
+}  // namespace ov
+
+// ! [executable_network:ctor_cnnnetwork]
+TemplatePlugin::CompiledModel::CompiledModel(const std::shared_ptr<const ov::Model>& model,
+                                             const std::shared_ptr<const ov::IPlugin>& plugin,
+                                             const InferenceEngine::ITaskExecutor::Ptr& task_executor,
+                                             const Configuration& cfg)
+    : ov::ICompiledModel(model, plugin, task_executor),  // Disable default threads creation
+      m_model(model->clone()),
+      _cfg(cfg) {
+    // TODO: if your plugin supports device ID (more that single instance of device can be on host machine)
+    // you should select proper device based on KEY_DEVICE_ID or automatic behavior
+    // In this case, _waitExecutor should also be created per device.
+    try {
+        compile_model(m_model);
+    } catch (const InferenceEngine::Exception&) {
+        throw;
+    } catch (const std::exception& e) {
+        IE_THROW(Unexpected) << "Standard exception from compilation library: " << e.what();
+    } catch (...) {
+        IE_THROW(Unexpected) << "Generic exception is thrown";
+    }
+}
+// ! [executable_network:ctor_cnnnetwork]
+
+// ! [executable_network:map_graph]
+// forward declaration
+void transform_model(const std::shared_ptr<ov::Model>& model);
+
+void TemplatePlugin::CompiledModel::compile_model(const std::shared_ptr<ov::Model>& model) {
+    // apply plugins transformations
+    transform_model(model);
+    // Generate backend specific blob mappings. For example Inference Engine uses not ngraph::Result nodes friendly name
+    // as inference request output names but the name of the layer before.
+    // TODO: Remove it
+    size_t idx = 0;
+    for (auto&& result : model->get_results()) {
+        const auto& input = result->input_value(0);
+        auto name = ov::op::util::get_ie_output_name(input);
```

Comment:
is this code with mapping still valid? I hope it's simplier now and we can drop this mapping

Reply:
Yes I need this code, because still work with legacy Infer request, it will be removed in future.